pkg/daemon/daemon.go

```diff
@@ -1629,3 +1641,104 @@ func forceFileExists() bool {
 
 	return false
 }
+
+// patchLabelsOnNode takes a map of key/value string pairs and patches them as labels on the node
+func (dn *Daemon) patchLabelsOnNode(labels map[string]interface{}) error {
+	var err error
+	var patchData []byte
+	patch := struct {
+		Metadata map[string]interface{} `json:"metadata"`
+	}{
+		Metadata: map[string]interface{}{
+			"labels": labels,
+		},
+	}
+	dn.logSystem("Patching labels %v", labels)
+	patchData, err = json.Marshal(&patch)
+	if err != nil {
+		dn.logSystem("Error in patching labels: %v", err)
+		return err
+	}
+	_, err = dn.kubeClient.CoreV1().Nodes().Patch(
+		context.TODO(), dn.name, types.MergePatchType, patchData, metav1.PatchOptions{})
+	if err != nil {
+		dn.logSystem("Error patching labels: %v", err)
+		return err
+	}
+	return nil
+}
+
+// Add or remove excludeFromLoadBalancerLabel to the node. If addLabel=true, it
+// adds the label if the label doesn't already exist and sets it to a custom value
+// so as to know that it was added by MCO. If addLabel=false, it removes the label,
+// only if its value is the known custom value, otherwise it leaves it as is.
+func (dn *Daemon) addOrRemoveExcludeFromLoadBalancerLabel(addLabel bool) error {
+	var err error
+	var labelValue interface{}
+	// Add the label only if not present; remove the label only if it has our custom value
+	labelValue, labelExists := dn.node.Labels[excludeFromLoadBalancerLabel]
```

Comment:
yeppp see:
https://github.com/openshift/machine-config-operator/blob/b68d8e1583e1b117e4fa968f0482e718d8d42f03/pkg/daemon/update.go#L356

you need to add a if dn.node{} around this conditional so it can pass in the firstboot where there is no node object.